src/engine/SCons/Node/Python.py

```diff
@@ -173,6 +176,23 @@ def get_csig(self, calc=None):
         self.get_ninfo().csig = contents
         return contents
 
+
+def ValueWithMemo(value, built_value=None):
+    global _memoLookupMap
+
+    # No current support for memoizing a value that needs to be built.
+    if built_value:
+        return Value(value, built_value)
+
+    value_str = str(value)
```

Comment:
Should this be wrapped with try/except in the case the value can't be converted to str ?

Reply:
I don't see the __str__ function using try/except so AFAICT this class assumes that the value can be converted to a string. I don't mind wrapping in try/except but I would want to add a test case. Do you know of any examples that I should try to support?

Reply:
If value is mutable then it shouldn't be cached right? (list, dict, other?)

Reply:
So you are saying that you want me to limit this to primitive types: int, bool, float, str?

Reply:
Ok I limited to primitive types and pushed an update.